com.wamas.ide.launching/src/com/wamas/ide/launching/generator/StandaloneLaunchConfigGenerator.xtend

```diff
@@ -125,16 +145,50 @@ class StandaloneLaunchConfigGenerator {
 			case LaunchConfigType.JAVA:
 				generateJava(config, copy)
 			case ECLIPSE:
-				generateEclipse(config, copy)
+				generateEclipse(config, copy, model)
 			case GROUP:
 				generateGroup(config, copy)
 			case RAP:
-				generateRAP(config, copy)
+				generateRAP(config, copy, model)
 		}
 
 		copy.doSave
 	}
-	
+
+	private def loadProductIfAvailable(LaunchConfig  config) {
+		val cp = config.collectContentProvider
+		if (cp === null || cp.empty) {
+			return null
+		}
+
+		val wsFile = cp.getWorkspaceFile
+		if (wsFile === null || !wsFile.exists) {
+			return null
+		}
+
+		val model = new WorkspaceProductModel(wsFile, false)
+		model.load
+
+		return model;
+	}
+
+	private def getWorkspaceFile(String cp) {
+		val file = new File(cp)
+		val wsRoot = ResourcesPlugin.workspace.root
+
+		if (file.absolute) {
+			val wsFiles = wsRoot.findFilesForLocationURI(file.toURI)
+			if (wsFiles === null || wsFiles.length <= 0) {
+				return null
+			}
+
+			return wsFiles.get(0)
+		}
+
+		val wsPath = wsRoot.location.append(cp)
```

Comment:
I don't see where adding support for relative paths helps. You can use any string substitution in the path anyway already, eg. "${workspace_loc}/path/to/product". Also, having files in the workspace directory is for instance when using EGit not even the default, as EGit by default clones repositories to the home directory...